metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1994,6 +2014,24 @@ private void listReassigningTopic(ListPartitionReassignmentsResponseData respons
             setReplicas(Replicas.toList(partition.replicas)));
     }
 
+    // Make it public to be visible in test.
+    public Integer getTopicEffectiveMinIsr(String topicName) {
+        Integer currentMinIsr = Integer.parseInt(configurationControl.getTopicConfigs(topicName)
+            .getOrDefault(MIN_IN_SYNC_REPLICAS_CONFIG, String.valueOf(defaultMinInSyncIsr)));
+        Integer replicationFactor = (int) defaultReplicationFactor;
+        try {
+            Uuid topicId = topicsByName.get(topicName);
+            replicationFactor = topics.get(topicId).parts.get(0).replicas.length;
+        } catch (Exception e) {
+            log.debug("Can't find the replication factor for topic: " + topicName);
+        }
+        return Math.min(currentMinIsr, replicationFactor);
+    }
+
+    private short partitionRecordVersion(boolean isElrAllowed) {
```

Comment:
This should be static _or_ it should compute the record version based on ClusterControl

Reply:
Removed, using the MV to determine the record version.
Even if the ELR static config is disabled, the partition records should be able to use the latest version in order not to block future features.